### PR TITLE
feat: logging search history to contents

### PR DESCRIPTION
**Pull request by hea9549**

### CODE REVIEW

File: src/main/java/orangetaxiteam/cocoman/application/ContentsApplicationService.java [hunk 1]
```diff
@@ -1,28 +1,41 @@
...
+    private final UserRepository userRepository;
 
     public ContentsApplicationService(
             ContentsRepository contentsRepository,
             GenreRepository genreRepository,
-            OttRepository ottRepository
-    ) {
+            OttRepository ottRepository,
+            SearchHistoryRepository searchHistoryRepository,
+            UserRepository userRepository) {
```
Comment (hihiboss) [inline src/main/java/orangetaxiteam/cocoman/application/ContentsApplicationService.java]:
```suggestion
            UserRepository userRepository
        ) {
```

File: src/main/java/orangetaxiteam/cocoman/application/ContentsApplicationService.java [hunk 2]
```diff
@@ -36,4 +49,32 @@ public ContentsDTO findById(String id) {
...
         );
     }
+
+    @Transactional(readOnly = true)
+    public List<ContentsDTO> findAll() {
+        return this.contentsRepository.findAll()
+                .stream()
+                .map(ContentsDTO::from)
+                .collect(Collectors.toList());
+    }
```
Comment (hihiboss) [inline src/main/java/orangetaxiteam/cocoman/application/ContentsApplicationService.java]: 지워주세요!

File: src/main/java/orangetaxiteam/cocoman/application/ContentsApplicationService.java [hunk 3]
```diff
@@ -36,4 +49,32 @@ public ContentsDTO findById(String id) {
...
+
+
+    @Transactional
+    public void createSearchHistory(String id, String keyword, String currentUserId) {
```
Comment (hihiboss) [inline src/main/java/orangetaxiteam/cocoman/application/ContentsApplicationService.java]:
```suggestion
    public void createSearchHistory(String contentId, String keyword, String currentUserId) {
```

File: src/main/java/orangetaxiteam/cocoman/domain/SearchHistory.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.time.LocalDateTime;
+
+@Entity
+@Getter
+@NoArgsConstructor
+@Table(name = "TB_SEARCH_HISTORY")
+@EntityListeners(AuditingEntityListener.class)
+public class SearchHistory {
```
Comment (hihiboss) [inline src/main/java/orangetaxiteam/cocoman/domain/SearchHistory.java]:
```suggestion
public class SearchHistory extends DomainEntity {
```

**Review by hihiboss — APPROVED** [review]

**Review by GunHee5719 — APPROVED** [review]
LGTM:)

**Pull request merged**